### PR TITLE
Collaborative code editor completed 

### PR DESCRIPTION
**Pull request by Gibson0918**
Collaborative code editor working as intended with basic syntax highlighting and auto-completion support. Auto-completion varies across language but should be reasonable given our use case of coding algorithms.

### CODE REVIEW

**Review by mslevis — APPROVED** [review]
LGTM

File: Frontend/src/pages/questions.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import Head from "next/head";
-import NavBar from '../components/nav'
+import NavBar from '../components/Nav'
```
Comment (shuimeihe) [inline Frontend/src/pages/questions.tsx:2]: a

File: Frontend/src/pages/users.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import Head from "next/head";
-import NavBar from '../components/nav'
+import NavBar from '../components/Nav'
```
Comment (shuimeihe) [inline Frontend/src/pages/users.tsx:2]: these 2 parts

**Review by shuimeihe — APPROVED** [review]

File: Frontend/src/pages/users.tsx (hunk 2 above)
Comment (shuimeihe) [reply Frontend/src/pages/users.tsx:2]: the component is still showing up as nav.tsx in the code review editor

if theres no errors please ignore my comments 🦖

**Pull request merged**